test/python/pulse/test_parameter_manager.py

```diff
@@ -352,10 +352,7 @@ def test_complex_valued_parameter(self):
         with self.assertWarns(PendingDeprecationWarning):
             assigned = visitor.visit(test_obj)
 
-        with self.assertWarns(DeprecationWarning):
-            ref_obj = pulse.Constant(duration=160, amp=1j * 0.1)
-
-        self.assertEqual(assigned, ref_obj)
+        self.assertEqual(assigned.amp, 0.1j)
```

Comment:
Seems like this is confusing because we can still use complex valued amplitude through parameterization (because this bypasses validation in the constructor). We can probably drop validation in the constructor and let user use valid amplitude value _on their responsibility_. Note that this pulse will fail in execution anyways because QPY doesn't take complex value, right? I think this test doesn't really make sense.

Reply:
@nkanazawa1989 

With Qiskit's main branch you can save a `ScheduleBlock` with complex `amp`.

In #9897 we added a `PendingDeprecationWarning` for the assignment of complex values. We can promote to deprecation in 0.46 and remove in 1.0 if you want.